### PR TITLE
Update Solana token transfer models and add prices

### PR DESCRIPTION
**Pull request by 0xBoxer**
Updated Solana token transfer models to improve consistency and performance across SPL, Token2022, and SOL transfers. Added price information to enhance token transfer data.

### CODE REVIEW

File: dbt_subprojects/solana/models/tokens/solana/tokens_solana_transfers.sql [hunk 1]
```diff
@@ -76,7 +126,55 @@ SELECT
...
 LEFT JOIN {{ ref('solana_utils_token_accounts') }} tk_d ON tk_d.address = tr.to_token_account
 WHERE 1=1
--- AND call_block_time > now() - interval '90' day --for faster CI testing
\ No newline at end of file
+{% if is_incremental() %}
+AND {{incremental_predicate('block_time')}}
+{% endif %}
+{% if not is_incremental() %}
+AND block_time > now() - interval '7' day
+{% endif %}
```
Comment (Hosuke) [inline dbt_subprojects/solana/models/tokens/solana/tokens_solana_transfers.sql:137]: Do we need to remove those `if not is_incremental()` clauses?

Comment (0xBoxer) [reply dbt_subprojects/solana/models/tokens/solana/tokens_solana_transfers.sql:137]: ye definitely, just still testing this is to make sure the trial runs don't time out

**Comment by 0xBoxer** [comment]
@0xRobin @jeff-dude this is good to go from a logic perspective, but the job has memory limit problems for 30 days.
Any hints from past experiences how we could solve this/move this forward?

**Comment by 0xRobin** [comment]
I would set the partition to `block_date` instead of `block_month`, that should already limit the data a bit if read by partition.
And we probably want to adapt the `delete+insert` merge strategy here as well? 
That should be interchangeable with the `merge` strategy without modifying the logic

File: dbt_subprojects/solana/models/tokens/solana/tokens_solana_transfers.sql [hunk 2]
```diff
@@ -4,25 +4,28 @@
         alias = 'transfers',
         materialized = 'incremental',
         file_format = 'delta',
-        incremental_strategy = 'merge',
+        incremental_strategy = 'delete+insert',
         partition_by = ['block_date'],
-        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')],
+        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_date')],
```
Comment (0xRobin) [inline dbt_subprojects/solana/models/tokens/solana/tokens_solana_transfers.sql:9]: @0xBoxer apparently the `DBT_INTERNAL_DEST` prefix is specific to the `merge` strategy. Just using `block_date` should work :pray:

Comment (0xBoxer) [reply dbt_subprojects/solana/models/tokens/solana/tokens_solana_transfers.sql]: aha, thanks 🙏

**Comment by 0xBoxer** [comment]
Due to a dbt version change, you need to merge main into your branch before the Github action runner runs successfully again.

**Comment by 0xBoxer** [comment]
CI runs for 30 days of history now.

This will modify the existing `solana_transfers` spell, we should have a quick think about that, but IMO this is how the model should have been like from the start and is more akin to our other `blockchain_transfers` spell, we should just bite the bullet and do the switch here.

The final output is currently defined as a view that would respond well to filtering on `token_version` or any `block_date` filter. 

querying the data via test schemas in dune seems fine to me, trino is really good at doing it's predicate pushdown thing.

 here's a matview of net transfers based on this: https://dune.com/queries/4198707

**Comment by 0xBoxer** [comment]
can probably be closed in favour of https://github.com/duneanalytics/spellbook/pull/7074

**Pull request closed without merging**